test/celeritas/field/FieldPropagator.test.cc

```diff
@@ -512,7 +515,7 @@ TEST_F(TwoBoxTest, electron_small_step)
         EXPECT_TRUE(result.boundary);
         EXPECT_TRUE(geo.is_on_boundary());
         EXPECT_VEC_SOFT_EQ(Real3({5, 0, 0}), geo.pos());
-        EXPECT_VEC_EQ(Real3({1, 0, 0}), geo.dir());
+        EXPECT_VEC_SOFT_EQ(Real3({1, 2 * delta, 0}), geo.dir());
```

Comment:
Is the test failed with `delta` intead of `2 * delta`? Why do we need soft_eq only for the y direction?

Reply:
This change was due to 83cca245c. Because I forgot to update the momentum, the `y` direction wasn't changed (either at all, or enough to be nonzero after normalization). The direction should generally always be a "soft" check since they're real numbers; I'm not sure why I had exact comparison in previously.

Reply:
> I'm not sure why I had exact comparison in previously

Because I added this test ;)